### PR TITLE
Unit test tempfile - add more test cases

### PR DESCRIPTION
**Pull request by stevenyvr987**
The two commits in this PR originates as a followup to issue #1432. I wanted to add a test case or two to detect the issue. In the act of doing this, I realize a few more test cases of a similar nature could be added. Also, I added a test to directly verify temp directory deletion.

The new test cases are mostly concerned with testing two successive calls of gettempdir or tempname, and when the calls are interrupted by deleting the temp directory, or by doing the other operation. For example,
1. gettempdir, gettempdir
2. gettempdir, deltempdir, gettempdir - 'internally' delete temp directory
3. gettempdir, lfs.rmdir, gettempdir - 'externally' delete temp directory
4. gettempdir, tempname, gettempdir

Test case 1 appears in the original unit test, which is a check for idempotency.
Test case 4 checks for any side effects between gettempdir and tempname.
Test case 3 checks for #1432 and is to be compared with test case 2. Test case 3 is written to succeed, although it should more naturally fail, if it wasn't for the performance hit that will occur (see comments in #1432).

A similar test sequence is repeated for tempname.

Roughly speaking, the first commit re-expresses the original unit test with shortcut aliases so there is less verbosity. The first commit also merges two of the test cases into one. The second commit adds the extra test cases.

### CODE REVIEW

**Comment by josephholsten** [comment]
what would be required to get this rebased and merged? are their any substantive changes needed?

**Comment by justinmk** [comment]
Probably doesn't need major changes, and it would be nice to have these tests.

File: test/unit/tempfile_spec.lua [hunk 1]
```diff
@@ -1,60 +1,147 @@
+local cimport, vim_init, ffi, to_cstr, eq, neq = (function()
+  local _ = require 'test.unit.helpers'
```
Comment (justinmk) [inline test/unit/tempfile_spec.lua]: this needs to be:

    local _ = require("test.unit.helpers")(after_each)

**Comment by justinmk** [comment]
We have more tempfile tests since this was added.

**Pull request closed without merging**